### PR TITLE
Move Dialogs to Radix

### PR DESCRIPTION
**Pull request by mozzius**
Adds Radix's `focus-scope` and `focus-guard` utilities to the dialogs and composer, so that they can compose properly with the dropdown menus. This means that dialogs and menus can now be nested, and the `Escape` key will work properly.

I also did the same to the composer, so now focus loops around when you get to the end

I also added nested sheet support to the menu component on native

# Test plan

1. Test the new `Dialog with menu in it` button in the Storybook (all platforms). Before these changes, this would've caused a recursion bug on firefox, so make sure that doesn't happen
2. Test the dialogs on web, esp. with the keyboard. They should act the same
3. Test the composer and the nested dialogs within it for regressions. again, should act the same but now with looping focus

### CODE REVIEW

File: src/view/com/util/UserBanner.tsx [hunk 1]
```diff
@@ -90,14 +93,11 @@ export function UserBanner({
 
   // setUserBanner is only passed as prop on the EditProfile component
   return onSelectNewBanner ? (
-    <EventStopper onKeyDown={false}>
+    <EventStopper onKeyDown={true}>
```
Comment (estrattonbailey) [inline src/view/com/util/UserBanner.tsx:96]: Not sure we even need this anymore

**Review by estrattonbailey — APPROVED** [review]
Beautiful

File: src/components/Dialog/index.web.tsx [hunk 2]
```diff
@@ -10,7 +10,9 @@ import {
 import Animated, {FadeIn, FadeInDown} from 'react-native-reanimated'
 import {msg} from '@lingui/macro'
 import {useLingui} from '@lingui/react'
-import {FocusScope} from '@tamagui/focus-scope'
```
Comment (haileyok) [inline src/components/Dialog/index.web.tsx:13]: Can we remove this now?

File: src/view/com/util/UserBanner.tsx (hunk 1 above)
Comment (mozzius) [reply src/view/com/util/UserBanner.tsx:96]: oh cool! I'll follow up with a PR to remove those, wanna test thoroughly first so won't do it here

**Pull request merged**